### PR TITLE
[LOGMGR-117] Fork Log4J ConosleAppender to simply handling of calls to Syetem.out

### PR DESCRIPTION
**Pull request by gaol**
Jira: https://issues.jboss.org/browse/LOGMGR-117
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1196228

The commit contains a fork **ConsoleAppender** from log4j, it uses
 <code>new PrintStream(new FileOutputStream(FileDescriptor.out), true)</code>
 instead of    <code>System.out</code>  for the Console output.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -105,6 +105,8 @@
                                         <exclude>org/apache/log4j/Hierarchy$*.class</exclude>
                                         <exclude>org/apache/log4j/Category.class</exclude>
                                         <exclude>org/apache/log4j/Category$*.class</exclude>
+                                        <exclude>org/apache/log4j/ConsoleAppender.class</exclude>
+                                        <exclude>org/apache/log4j/ConsoleAppender$*.class</exclude>
```
Comment (gaol) [inline pom.xml]: Exclude forked ConsoleAppender* classes

File: src/test/java/org/apache/log4j/LoggerTest.java [hunk 2]
```diff
@@ -499,6 +501,30 @@ public void testAppenderEvent4() {
...
+            //consoleAppender.setFollow(true);
+            consoleAppender.setTarget("System.out");
+            consoleAppender.activateOptions();
+            logger.addAppender(consoleAppender);
+            consoleAppender.setLayout(new SimpleLayout());
+            logger.warn("hello console appender");
+
+            // ConsoleAppender redirects to real stdout, so nothing in the arrayOut
+            // this is failed before ConsoleAppender changes.
+            assertEquals("", arrayOut.toString());
```
Comment (gaol) [inline src/test/java/org/apache/log4j/LoggerTest.java]: This will fail if revert the changes in ConsoleAppender.

I don't have good way to test that the log message is redirected to real <code>stdout</code>, any idea about that?

**Comment by jamezp** [comment]
Is there a reason not to just use `System.out`? Also I don't think we really need a test in this case.

**Comment by jamezp** [comment]
This was all I had done with my testing and it seemed to work fine, https://github.com/jboss-logging/log4j-jboss-logmanager/compare/master...jamezp:console-appender

**Comment by gaol** [comment]
@jamezp During JBoss server starts up, the **System.out** has been reassigned by **StdioContext** to   **DelegatingPrintStream**. Thus later invocation of "System.out" is not the real **stdout** any more.

The ConsoleAppender class maybe loaded during deployment time, the **System.out** has been reassigned already at this time, so it should not be the real **stdout** in this situation.

**Comment by jamezp** [comment]
Ah, yeah. Usually we just do something like `Class.forName("org.apache.log4j.ConsoleAppender")` so it does the static init. I'll give it some thought, thanks!

**Comment by gaol** [comment]
I can reproduce the deadlock if I use:
<code>private static final PrintStream out = System.out</code> in **ConsoleAppender**

Reproducer war is located at bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1196228,
and I tested using latest EAP 6.4.x branch build

**Comment by gaol** [comment]
@jamezp  Welcome :-)

File: src/main/java/org/apache/log4j/ConsoleAppender.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+ * @author Curt Arnold
+ * @author <a href="mailto:lgao@redhat.com">Lin Gao</a>
+ * @since 1.1 */
+public class ConsoleAppender extends WriterAppender {
+
+  public static final String SYSTEM_OUT = "System.out";
+  public static final String SYSTEM_ERR = "System.err";
+
+  private static final PrintStream out = new PrintStream(new FileOutputStream(FileDescriptor.out), true);
+  private static final PrintStream err = new PrintStream(new FileOutputStream(FileDescriptor.err), true);
```
Comment (jamezp) [inline src/main/java/org/apache/log4j/ConsoleAppender.java]: These should probably be `new PrintStream(new BufferedOuputStream(new FileOutputStream(FileDescriptor.xxx), 128));` to keep in sync what the JDK does.

**Comment by jamezp** [comment]
I think we should just remove the test too.

**Comment by gaol** [comment]
@jamezp I updated as required and removed the test.
